### ISSUE TITLE
Possible bug in emmeans and Surv object

### ISSUE THREAD

**Issue by OnofriAndreaPG** [rvlenth/emmeans#52]
I am fitting a parametric survival model with survreg and I would like to get the means by using the emmeans() function. The function used to work fine until emmeans vers. 1.2.3, but it no longer works with vers. 1.2.4. I am getting the following error:

```
Error in parse(text = x, keep.source = FALSE) :    <text>:1:20: unexpected input 1: ~Surv(L, U, type = _
                       ^
```

My code is as follows:

```
BBsurvey <- structure(data.frame(1:36, structure(c(1L, 1L, 1L, 1L, 2L, 2L, 2L, 
2L, 3L, 3L, 3L, 3L, 4L, 4L, 4L, 4L, 5L, 5L, 5L, 5L, 6L, 6L, 6L, 
6L, 7L, 7L, 7L, 7L, 8L, 8L, 8L, 8L, 9L, 9L, 9L, 9L), .Label = c("A", 
"B", "C", "D", "E", "F", "G", "H", "I"), class = "factor"), c(0.1, 
0.1, 5, 5, 0.1, 0.1, 5, 5, 0, 0.1, 5, 5, 0, 0.1, 5, 5, 0, 0, 
0.1, 0.1, 5, 5, 25, 25, 0.1, 0.1, 5, 5, 0, 0.1, 5, 5, 75, 25, 
50, 25), c(5, 5, 25, 25, 5, 5, 25, 25, 0.1, 5, 25, 25, 0.1, 5, 
25, 25, 0.1, 0.1, 5, 5, 25, 25, 50, 50, 5, 5, 25, 25, 0.1, 5, 
25, 25, 100, 50, 75, 50), c(2.55, 2.55, 15, 15, 2.55, 2.55, 15, 
15, 0.05, 2.5, 15, 15, 0.05, 2.55, 15, 15, 0.05, 0.05, 2.55, 
2.55, 15, 15, 37.5, 37.5, 2.5, 2.55, 15, 15, 0.1, 2.55, 15, 15, 
87.5, 37.5, 62.5, 37.5)), .Names = c("X", "Herbicide", "L", "U", 
"midPoint"))

mod.surv <- survreg(Surv(L, U, type="interval2") ~ Herbicide, 
  dist="gaussian", data = BBsurvey)
means.surv <- emmeans(mod.surv, Herbicide)
```

Any help with this is very much appreciated. Thank you very much in advance.

Andrea

**Comment by rvlenth** [comment]
Thanks for reporting this. re you saying it *did* work with version 1.2.3, or that it *stopped* working with 1.2.3?

**Comment by OnofriAndreaPG** [comment]
Thanks for your answer. It did work with 1.2.3. With 1.2.4 I am getting the error message detailed in my previous email.

**Comment by rvlenth** [comment]
OK, I found the problem. It was a very subtle error; in case you want to know, it was caused by adding support for models with formulas like `data1[["resp"]] ~ data2[["x1"]]`, which in turn involves substituting oddball strings for `[[`,`"`,and`]]`. Thus, `type = "interval2"` became `type = _Av3_interval2_Av3_`, which cannot be parsed because it is illegal for a variable name to start with `_`. I changed my oddball strings and now it works.

But another thing popped up that I think must have happened before with this example, and that is that in the end, it thinks we are using a transformation named `Surv`. I fixed that too, I think.

I'll push it to github in just a little bit.
